.ci/job/job_runner.sh

```diff
@@ -0,0 +1,131 @@
+#!/bin/bash
+#
+# Copyright (c) 2021 Red Hat, Inc.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+cidir="$(dirname $0)/../"
+source "$cidir/lib.sh"
+
+#
+# Variables that the user can overwrite
+#
+KATA_TESTS_JOB_RUNNER="${KATA_TESTS_JOB_RUNNER:-vm}"
+
+#
+# Script variables
+#
+
+ALL_AVAILABLE_JOBS=(
+	"baremetal-pmem"
+	"cri_containerd_k8s"
+	"cri_containerd_k8s_initrd"
+	"cri_containerd_k8s_complete"
+	"cri_containerd_k8s_minimal"
+	"crio_k8s"
+	"crio_k8s_complete"
+	"crio_k8s_minimal"
+	"cloud-hypervisor-k8s-crio"
+	"cloud-hypervisor-k8s-containerd"
+	"cloud-hypervisor-k8s-containerd-minimal"
+	"Ccloud-hypervisor-k8s-containerd-fullL"
+	"firecracker"
+	"vfio"
+	"virtiofs_experimental"
+	"metrics"
+	"metrics_experimental")
+
+usage() {
+	cat <<-EOF
+	Run a CI job.
+	User can select the runner which will trigger the job within a given environment.
+
+	Usage: $0 [-h] [-l] [-j JOB_ID]
+
+	Options:
+	       -l: list all defined jobs
+	       -j JOB_ID: run the job of JOB_ID
+
+	Environment variables:
+	        KATA_TESTS_JOB_RUNNER: switch between 'local' and 'vm' (default) runners"
+	EOF
+}
+
+list_jobs_id() {
+	echo ${ALL_AVAILABLE_JOBS[@]}
+}
+
+#
+# Note:
+#
+# Any function named as _do_run_NAME is interpreted as the implementation of
+# the NAME runner. So if you want to add new runners then you just need to
+# create a function with that name pattern.
+#
+
+# Local runner implementation.
+_do_run_local() {
+	local job_id="$1"
+	local runner_script="${cidir}/run.sh"
+	local setup_script="${cidir}/setup.sh"
+	# Convert the ID to uppercase.
+	typeset -u CI_JOB="$job_id"
+	# Export it.
+	typeset -x CI_JOB
+	bash "$setup_script"
+	bash "$runner_script"
```

Comment:
Enforcing running the setup and runner immediately after each other worries me a bit.  Yes, that's currently the safe way to run the CI jobs, but
 - The setup script takes rather a long time
 - It means the setup and job output will be combined together, perpetuating Kata's existing problem of having huge logs where it's hard to figure out where things went wrong.